monitor-function-calls/src/agent.js

```
@@ -19,66 +22,71 @@ function createAlert(
 
-    // get the contract names that have events that we wish to monitor
     const contractNames = Object.keys(data.contractInfo);
 
-    // load relevant information for each contract
     data.contracts = contractNames.map((name) => {
       const { address, abiFile, functions = {} } = data.contractInfo[name];
       const abi = getAbi(abiFile);
       const iface = new ethers.utils.Interface(abi);
       const functionNames = Object.keys(functions);
 
-      // attempt to get function signatures for each of the function names in the config file
-      let functionSignatures = functionNames.map(
-        (functionName) => {
-          try {
-            const fragment = iface.getFunction(functionName);
-            return fragment.format(ethers.utils.FormatTypes.full);
-          } catch {
-            // ignore error thrown by ethers if it cannot find a suitable function fragment and
-            // return an empty string
-            return '';
-          }
-        },
-      );
+      let functionSignatures = functionNames.map((functionName) => {
+        const { expression, type, severity } = functions[functionName];
+        const fragment = iface.getFunction(functionName);
 
-      // filter out any emtpy strings from functionSignatures, an empty string denotes that
-      // attempting to get the signature for a requested function name failed
-      functionSignatures = functionSignatures.filter((signature) => signature !== '');
+        const result = {
+          functionName,
+          signature: fragment.format(ethers.utils.FormatTypes.full),
+          functionType: type,
+          functionSeverity: severity,
+        };
+
+        if (expression !== undefined) {
+          result.expression = expression;
+          result.expressionObject = parseExpression(expression);
+        }
+
+        return result;
+      });
+
+      functionSignatures = functionSignatures.filter((result) => result !== '');
```

Since the `try/catch` around `getFunction` has been removed, there's no need to filter here since it will just throw instead of adding an empty string to the `functionSignatures` array

---

Fixed.